lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -293,16 +296,58 @@ instance Buildable (PartialTx era) where
         cardanoTxF :: Cardano.Tx era -> Builder
         cardanoTxF tx' = pretty $ pShow tx'
 
+-- validateUTxO :: UTxO -> TransactionLayer -> ...
+--
+data UTxOAssumptions = forall k ktype. UTxOAssumptions
+    { txLayer
+        :: TransactionLayer k ktype SealedTx
+        -- TODO: Replace with smaller and smaller parts of 'TransactionLayer'
+    , inputScriptLookup
+        :: Maybe ([(W.TxIn, W.TxOut)] -> [CA.Script KeyHash])
+    , inputScriptTemplate
+        :: Maybe ScriptTemplate
+    , description :: String
+    }
+
+instance Show UTxOAssumptions where
+    show (UTxOAssumptions _ _ template description) = unwords
+        [ "UTxOAssumptions {"
+        , "description = " <> show description
+        , "template = " <> show template
+        , "}"
+        ]
+
+-- Coin-selection assuming wallet UTxOs are either
+allKeyPaymentCredentials
+    :: forall k. TransactionLayer k 'CredFromKeyK SealedTx
+    -> UTxOAssumptions
+allKeyPaymentCredentials tl =
+    UTxOAssumptions
+        tl
+        Nothing
+        Nothing
+        "allKeyPaymentCredentials"
```

Comment:
Perhaps use ordinary record construction here, instead of positional construction?

With ordinary record construction, it's easier for readers to see which field corresponds to which value.

Reply:
Done